error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/BigDecimalRulesTestInput.java

```diff
@@ -20,4 +20,36 @@ ImmutableSet<BigDecimal> testBigDecimalTen() {
   ImmutableSet<BigDecimal> testBigDecimalValueOf() {
     return ImmutableSet.of(new BigDecimal(2), new BigDecimal(2L), new BigDecimal(2.0));
   }
+
+  ImmutableSet<Boolean> testBigDecimalSignumIsZero() {
+    return ImmutableSet.of(
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) == 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) == 0,
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) != 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) != 0);
+  }
+
+  ImmutableSet<Boolean> testBigDecimalSignumIsPositive() {
+    return ImmutableSet.of(
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) > 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) < 0,
+        BigDecimal.ZERO.signum() == 1,
+        BigDecimal.ZERO.signum() >= 1,
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) <= 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) >= 0,
+        BigDecimal.ZERO.signum() != 1,
+        BigDecimal.ZERO.signum() < 1);
+  }
+
+  ImmutableSet<Boolean> testBigDecimalSignumIsNegative() {
+    return ImmutableSet.of(
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) < 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) > 0,
+        BigDecimal.ZERO.signum() == -1,
+        BigDecimal.ZERO.signum() <= -1,
+        BigDecimal.ONE.compareTo(BigDecimal.ZERO) >= 0,
+        BigDecimal.ZERO.compareTo(BigDecimal.ONE) <= 0,
+        BigDecimal.ZERO.signum() != -1,
```

Comment:
```suggestion
        BigDecimal.ZERO.signum() != 1,
```

Reply:
This test case is for the negation :)

Reply:
Guess I don't really follow these last 2 test cases (neither above), as the test titles seem to suggest these are variants of checking if a number is **negative**, but the last two actually do the reverse (and same above)

Reply:
> as the test titles seem to suggest these are variants of checking if a number is *negative*, but the last two actually do the reverse (and the same applies above).

Correct; the second half of the test cases cover the implicitly-generated `@AlsoNegation` rules.

> Probably the `ZERO` and `ONE` don't really help here either.

Yeah, we generally try to vary the captured sub-expressions; lemme try to push something.

Reply:
Pushed something.

Reply:
> the implicitly-generated @AlsoNegation rules

Ah right, that makes it a bit harder. Varying the numbers helps a bit though 👍🏻